packages/devtools_app/lib/src/screens/memory/panes/chart/chart_pane.dart

```diff
@@ -457,25 +455,15 @@ class _MemoryChartPaneState extends State<MemoryChartPane>
       }
 
       Widget traceColor;
-      if (colorPatch != null) {
-        if (dashed) {
-          traceColor = createDashWidget(colorPatch);
-        } else {
-          traceColor = createSolidLine(colorPatch);
-        }
-      } else {
-        traceColor = image == null
-            ? const SizedBox()
-            : scaleImage
-                ? Image(
-                    image: AssetImage(image),
-                    width: 20,
-                    height: 10,
-                  )
-                : Image(
-                    image: AssetImage(image),
-                  );
-      }
+      traceColor = colorPatch != null
+          ? dashed
+              ? createDashWidget(colorPatch)
+              : createSolidLine(colorPatch)
+          : image == null
+              ? const SizedBox()
+              : scaleImage
+                  ? Image(image: AssetImage(image), width: 20, height: 10)
+                  : Image(image: AssetImage(image));
```

Comment:
triple nest 😱

Reply:
yeah. if we were to add a lint about ternary operators, it would never suggest more than one nesting level and never suggest large blocks.